File: app/src/main/java/com/wkimdev/mytripnote/adapter/NoteAdapter.java
package com.wkimdev.mytripnote.adapter;

import android.content.Context;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;
import android.widget.ImageView;
import android.widget.LinearLayout;
import android.widget.TextView;

import androidx.annotation.NonNull;
import androidx.recyclerview.widget.RecyclerView;

import com.bumptech.glide.Glide;
import com.wkimdev.mytripnote.model.NoteItem;
import com.wkimdev.mytripnote.OnClickListener;
import com.wkimdev.mytripnote.R;

import java.util.ArrayList;

/**
 * 일반 여행 노트 리스트 리사이클러뷰 어댑터
 */
public class NoteAdapter extends RecyclerView.Adapter<NoteAdapter.NoteAdapterViewHolder>{

    private ArrayList<NoteItem> noteItems;
    private OnClickListener onClickListener;
    private Context context;

    public NoteAdapter(ArrayList<NoteItem> noteItems, Context context, OnClickListener onClickListener) {
        this.noteItems = noteItems;
        this.context = context;
        this.onClickListener = onClickListener;
    }

    @NonNull
    @Override
    public NoteAdapterViewHolder onCreateViewHolder(@NonNull ViewGroup parent, int viewType) {
        View view = LayoutInflater
                .from(parent.getContext())
                .inflate(R.layout.item_note, parent, false);

        return new NoteAdapterViewHolder(view);
    }

    @Override
    public void onBindViewHolder(@NonNull NoteAdapterViewHolder holder, int position) {

        holder.tv_note_title.setText(noteItems.get(position).getNoteTitle());
        holder.tv_note_content.setText(noteItems.get(position).getNoteContent());
        Glide.with(context).load(noteItems.get(position).getYoutubeThumnailImage()).into(holder.iv_youtube_thumnail);

        holder.itemView.setTag(position);

        holder.layout_note.setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View v) {
                //id를 기준으로 던져서 조회 후 호출
                onClickListener.onClickNoteList(noteItems.get((int)holder.itemView.getTag()));
            }
        });

    }

    @Override
    public int getItemCount() {
        return ( null != noteItems ? noteItems.size() : 0);
    }

    class NoteAdapterViewHolder extends RecyclerView.ViewHolder {

        private TextView tv_note_title;
        private TextView tv_note_content;
        private ImageView iv_youtube_thumnail;
        private LinearLayout layout_note;

        public NoteAdapterViewHolder(@NonNull View itemView) {
            super(itemView);

            this.tv_note_title = itemView.findViewById(R.id.tv_note_title);
            this.tv_note_content = itemView.findViewById(R.id.tv_note_content);
            this.iv_youtube_thumnail = itemView.findViewById(R.id.iv_youtube_thumnail);
            this.layout_note = itemView.findViewById(R.id.layout_note);
        }
    }

}

File: app/src/main/java/com/wkimdev/mytripnote/adapter/NoteAdapter.kt
package com.wkimdev.mytripnote.adapter

import android.content.Context
import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import android.widget.ImageView
import android.widget.LinearLayout
import android.widget.TextView
import androidx.recyclerview.widget.RecyclerView
import com.bumptech.glide.Glide
import com.wkimdev.mytripnote.OnClickListener
import com.wkimdev.mytripnote.R
import com.wkimdev.mytripnote.adapter.NoteAdapter.NoteAdapterViewHolder
import com.wkimdev.mytripnote.model.NoteItem

/**
 * 일반 여행 노트 리스트 리사이클러뷰 어댑터
 */
class NoteAdapter(
    private val noteItems: ArrayList<NoteItem>?,
    private val context: Context,
    private val onClickListener: OnClickListener
) : RecyclerView.Adapter<NoteAdapterViewHolder>() {
    override fun onCreateViewHolder(parent: ViewGroup, viewType: Int): NoteAdapterViewHolder {
        val view = LayoutInflater
            .from(parent.context)
            .inflate(R.layout.item_note, parent, false)
        return NoteAdapterViewHolder(view)
    }

    override fun onBindViewHolder(holder: NoteAdapterViewHolder, position: Int) {
        holder.tv_note_title.text = noteItems!![position].noteTitle
        holder.tv_note_content.text = noteItems[position].noteContent
        Glide.with(context).load(noteItems[position].youtubeThumnailImage)
            .into(holder.iv_youtube_thumnail)
        holder.itemView.tag = position
        holder.layout_note.setOnClickListener(View.OnClickListener { //id를 기준으로 던져서 조회 후 호출
            onClickListener.onClickNoteList(noteItems[holder.itemView.tag as Int])
        })
    }

    override fun getItemCount(): Int {
        return noteItems?.size ?: 0
    }

    inner class NoteAdapterViewHolder(itemView: View) : RecyclerView.ViewHolder(itemView) {
        val tv_note_title: TextView
        val tv_note_content: TextView
        val iv_youtube_thumnail: ImageView
        val layout_note: LinearLayout

        init {
            tv_note_title = itemView.findViewById(R.id.tv_note_title)
            tv_note_content = itemView.findViewById(R.id.tv_note_content)
            iv_youtube_thumnail = itemView.findViewById(R.id.iv_youtube_thumnail)
            layout_note = itemView.findViewById(R.id.layout_note)
        }
    }
}